rd-net/Lifetimes/Lifetimes/LifetimeDefinition.cs

```diff
@@ -499,12 +499,12 @@ private void MarkCancelingRecursively()
     private void Destruct()
     {
       var status = Status;
-      Assertion.Assert(status == LifetimeStatus.Terminating, "{0}: bad status for destructuring start", this);
-      Assertion.Assert(ourMutexSlice[myState] == false, "{0}: mutex must be released in this point", this);
+      if (Mode.IsAssertion) Assertion.Assert(status == LifetimeStatus.Terminating, "{0}: bad status for destructuring start", this);
+      if (Mode.IsAssertion) Assertion.Assert(ourMutexSlice[myState] == false, "{0}: mutex must be released in this point", this);
       //no one can take mutex after this point
 
       var resources = myResources;
-      Assertion.Assert(resources != null, "{0}: `resources` can't be null on destructuring stage", this);
+      if (Mode.IsAssertion) Assertion.Assert(resources != null, "{0}: `resources` can't be null on destructuring stage", this);
```

Comment:
_(Minor)_ Let's add `!` below, in the `try-catch` block, to shut up the inspection?

Reply:
Thanks, fixed!